src/Raven.Server/RavenServer.cs

```diff
@@ -929,45 +935,188 @@ internal AuthenticateConnection AuthenticateConnectionCertificate(X509Certificat
+                            var certsWithSameHash = ServerStore.Cluster.GetCertificatesByPinningHashSortedByExpiration(ctx, pinningHash);
+
+                            if (certsWithSameHash.Count > 0)
+                            {
+                                // Hash is good, let's validate it was signed by same issuer, otherwise users can use the private key to register a new cert with a different issuer.
+                                if (CertHasKnownIssuer(certificate, certsWithSameHash) == false)
+                                {
+                                    if (Logger.IsOperationsEnabled)
+                                        Logger.Operations($"You tried to use a certificate: '{certificate.Subject} ({certificate.Thumbprint})' which is not " +
+                                                          "registered in the cluster explicitly but is trusted implicitly by its HTTP Public Key Pinning Hash (HPKP). " +
+                                                          "Your certificate was signed by an unknown issuer. This is not allowed - closing the connection. " +
+                                                          "To fix this, the admin can register the thumbprint of the *issuer* certificate (e.g. Let's Encrypt) in the 'Security.WellKnownIssuers.Admin' configuration entry.");
+
+                                    authenticationStatus.Status = AuthenticationStatus.UnfamiliarCertificate;
+                                    return authenticationStatus;
+                                }
+
+                                // Success, we'll add the new certificate with same permissions as the original
+                                var existingCert = certsWithSameHash[0];
+                                var newCertBytes = certificate.Export(X509ContentType.Cert);
+
+                                var newCertDef = new CertificateDefinition()
+                                {
+                                    Name = existingCert.Name,
+                                    Certificate = Convert.ToBase64String(newCertBytes),
+                                    Permissions = existingCert.Permissions,
+                                    SecurityClearance = existingCert.SecurityClearance,
+                                    Password = existingCert.Password,
+                                    Thumbprint = certificate.Thumbprint,
+                                    PublicKeyPinningHash = pinningHash,
+                                    NotAfter = certificate.NotAfter
+                                };
+                                
+                                // This command will discard leftover certificates after the new certificate is saved.
+                                ServerStore.SendToLeaderAsync(new PutCertificateWithSamePinningHashCommand(certKey, newCertDef))
+                                    .Wait(ServerStore.ServerShutdown);
```

Comment:
This isn't correct. We are now waiting for the cluster to go through.
We _must_ be able to proceed normally even without the cluster.

Note that the cluster command need to handle the case of repeated registration of the same cert.

Reply:
done